tests/schema_registry/test_avro_serializer.py

```diff
@@ -176,3 +176,66 @@ def test_avro_serializer_subject_name_strategy_default(load_avsc):
     ctx = SerializationContext('test_subj', MessageField.VALUE)
     assert test_serializer._subject_name_func(
         ctx, test_serializer._schema_name) == 'test_subj-value'
+
+
+def test_serialization_context_receives_headers(load_avsc):
+    """
+    Ensures SerializationContext headers parameter is properly stored
+    """
+    conf = {'url': TEST_URL}
+    test_client = SchemaRegistryClient(conf)
+    test_serializer = AvroSerializer(test_client, 'int',
+                                     conf={'subject.name.strategy':
+                                           record_subject_name_strategy})
+
+    ctx = SerializationContext('test_subj', MessageField.VALUE, [('header1', 'value1'),])
+    assert ('header1', 'value1') in ctx.headers
+
+
+def test_serialization_context_headers_optional(load_avsc):
+    """
+    Ensures SerializationContext headers parameter is optional
+    """
+    conf = {'url': TEST_URL}
+    test_client = SchemaRegistryClient(conf)
+    test_serializer = AvroSerializer(test_client, 'int',
+                                     conf={'subject.name.strategy':
+                                           record_subject_name_strategy})
+
+    ctx = SerializationContext('test_subj', MessageField.VALUE)
+    assert ctx is not None
+    assert ctx.headers is None
+
+
+def test_serialization_context_headers_empty(load_avsc):
+    """
+    Ensures SerializationContext headers parameter can be empty
+    """
+    conf = {'url': TEST_URL}
+    test_client = SchemaRegistryClient(conf)
+    test_serializer = AvroSerializer(test_client, 'int',
+                                     conf={'subject.name.strategy':
+                                           record_subject_name_strategy})
+
+    ctx = SerializationContext('test_subj', MessageField.VALUE, [])
+    assert ctx is not None
+    assert not ctx.headers
+
+
+def test_serialization_context_headers_updated(load_avsc):
+    """
+    Ensures SerializationContext headers parameter can be updated
+    """
+    conf = {'url': TEST_URL}
+    test_client = SchemaRegistryClient(conf)
+    test_serializer = AvroSerializer(test_client, 'int',
+                                     conf={'subject.name.strategy':
+                                           record_subject_name_strategy})
+
+    headers = [('k1', 'v1'),]
+
+    ctx = SerializationContext('test_subj', MessageField.VALUE, headers)
+    assert ('k1', 'v1') in ctx.headers
+    headers.extend([('k2', 'v2'),])
+    assert ('k2', 'v2') in ctx.headers
+    assert len(headers) == 2
```

Comment:
I prefer as exact as possible matching in tests, and order matters for the headers, so suggest doing an exact match between ctx.headers and what is expected in there.